torch/_subclasses/fake_tensor.py

```diff
@@ -572,9 +583,10 @@ def __set__(self, obj: FakeTensor, value: Optional[torch.SymInt]) -> None:
             setattr(obj, self._memo(obj), None)
             setattr(obj, self._memo_vc(obj), None)
             setattr(obj, self._memo_epoch(obj), None)
-        elif not torch.is_inference_mode_enabled():
+        elif not torch.is_inference_mode_enabled() or self._opt_vc:
```

Comment:
I wonder why we don't just check `obj.is_inference()` here

Reply:
Yeah, is inference seems better